### ISSUE TITLE
Turning off GPC signal from popup page automatically logs user out from their Google / Github Account

### ISSUE THREAD

**Issue by chunyuema** [privacy-tech-lab/optmeowt-browser-extension#181]
Currently, attempting to turn off the GPC signal through the popup will automatically logs the user out from Google/Github account that they are currently logged in. Based on my inspection of the code base, it appears to be due to fact that turning off the signal invokes `removeFromDomainlist` function, which in turn invokes `deleteDomainCookies` function. The `deleteDomainCookies` function is probably not implemented properly and accidentally deletes the user login information, thus resulting in the bug. 

A quick fix is by commenting out the `deleteDomainCookies` from `removeFromDomainlist` and we can observe that the bug no longer happens. However, I would imagine that it defeats the purpose of `deleteDomainCookies`. I am personally not familiar with what  `deleteDomainCookies` is trying to do so maybe someone who is more familiar with this code base can take a look and see if we can fix the bug by changing `deleteDomainCookies`.

Adding @kalicki1 and @SebastianZimmeck to this thread as suggested by the developer who currently works on refactoring of the extension.

**Comment by kalicki1** [comment]
Thanks @chunyuema, this is a great catch! Yes, I remember we had discussed this earlier in the year. If I remember correctly, `deleteDomainCookies` should remove the opt-out cookies associated with a domain, since our intention with the extension is to signal a user's opt-out intent via all possible methods, i.e., if someone opts-out, we should remove all opt-out signals.

Now that this is on our minds we can take a look at which cookies `deleteDomainCookies` deletes. I can take a look later today to see if I can catch this by glimpsing at the code when I have a moment.

**Comment by SebastianZimmeck** [comment]
Great, indeed, @chunyuema!

**Comment by kalicki1** [comment]
I took a second look at this code and I think we need to find a way to save which cookies preserve your login session. That way we can wipe all of the cookies placed on a domain that is logged out (it will be hard to record all opt-out cookies that exist) except for those. This is because the `deleteDomainCookies` function just deletes all of those cookies associated with a domain. 

Alternatively we can iterate over those cookies that we place and remove those. 

These are the two ways I see of implementing a fix for this. I will wait until the refactor to implement something along these lines.

**Comment by kalicki1** [comment]
Looking this over again at this point in time, I think we will either need to discuss this or keep this open for a little bit longer. Since this is related to deleting all cookies of a domain, this is still an issue because we actually have not reimplemented the functionality that deletes all the Do-Not-Sell cookies off a domain when a user switches Do-Not-Sell off for a domain. As a result, if we ever reimplement this functionality we still need abide by my comments from above. 

Let's discuss today whether or not we want to fully reimplement that or not. That will determine what we do here.

**Comment by SebastianZimmeck** [comment]
Very good observation, @kalicki1!

**Comment by kalicki1** [comment]
I will go ahead and reimplement this functionality as it was prior to refactoring as discussed in today's meeting.

**Comment by kalicki1** [comment]
I just pushed some changes to reimplement this functionality. Everything seems to be working as intended; I added a check whether something is being modified in the domainlist right into `storage.js`, which calls functions to carry out setting/removing the cookies via `storageCookies.js`'s exports. 

As a side note related to our IAB cookie implementation, at the moment the IAB cookie for a domain is removed only when all of the 3rd party domains, in addition to the 1st party domain, are chosen to not receive do not sell signals. Regardless of whether the 1st part is off or on, the one IAB cookie is still set to the 1st party domain regardless.
